src/tests.rs

```diff
@@ -102,25 +104,41 @@ macro_rules! gen_tests {
+
+    (@serde $val0:expr, $ty:ty, true) => {
+        #[cfg(feature="serde")]
+        use bincode;
+
+        #[cfg(feature="serde")]
+        #[test]
+        fn test_serde_round_trip() {
+            let src = Atomic::new($val0);
+            let serialized = bincode::serialize(&src).unwrap();
+            let deserialized: Atomic<$ty> = bincode::deserialize(&serialized).unwrap();
+            assert_eq!(src.load(Ordering::SeqCst), deserialized.load(Ordering::SeqCst));
+        }
+    };
+
+    (@serde $val0:expr, $ty:ty, false) => {};
 }
 
-//         mod     ty     val0    val1     logic  int    default
-gen_tests!(_bool,  bool,  true,   false,   true,  false, true);
-gen_tests!(_u8,    u8,    7u8,    33u8,    true,  true,  true);
-gen_tests!(_i8,    i8,    7i8,    33i8,    true,  true,  true);
-gen_tests!(_u16,   u16,   7u16,   33u16,   true,  true,  true);
-gen_tests!(_i16,   i16,   7i16,   33i16,   true,  true,  true);
-gen_tests!(_u32,   u32,   7u32,   33u32,   true,  true,  true);
-gen_tests!(_i32,   i32,   7i32,   33i32,   true,  true,  true);
-gen_tests!(_u64,   u64,   7u64,   33u64,   true,  true,  true);
-gen_tests!(_i64,   i64,   7i64,   33i64,   true,  true,  true);
-gen_tests!(_usize, usize, 7usize, 33usize, true,  true,  true);
-gen_tests!(_isize, isize, 7isize, 33isize, true,  true,  true);
-gen_tests!(_f32,   f32,   7.0f32, 33.0f32, false, false, true);
-gen_tests!(_f64,   f64,   7.0f64, 33.0f64, false, false, true);
-gen_tests!(_char,  char,  'x',    '♥',     false, false, true);
-gen_tests!(_ptr, *mut String, 0 as *mut String, 0xBADC0DE as *mut String, false, false, false);
-gen_tests!(custom, super::Foo, super::Foo::Nothing, super::Foo::Set(0b101), false, false, true);
+//         mod     ty     val0    val1     logic  int    dflt  serde
+gen_tests!(_bool,  bool,  true,   false,   true,  false, true, true);
+gen_tests!(_u8,    u8,    7u8,    33u8,    true,  true,  true, true);
+gen_tests!(_i8,    i8,    7i8,    33i8,    true,  true,  true, true);
+gen_tests!(_u16,   u16,   7u16,   33u16,   true,  true,  true, true);
+gen_tests!(_i16,   i16,   7i16,   33i16,   true,  true,  true, true);
+gen_tests!(_u32,   u32,   7u32,   33u32,   true,  true,  true, true);
+gen_tests!(_i32,   i32,   7i32,   33i32,   true,  true,  true, true);
+gen_tests!(_u64,   u64,   7u64,   33u64,   true,  true,  true, true);
+gen_tests!(_i64,   i64,   7i64,   33i64,   true,  true,  true, true);
+gen_tests!(_usize, usize, 7usize, 33usize, true,  true,  true, true);
+gen_tests!(_isize, isize, 7isize, 33isize, true,  true,  true, true);
+gen_tests!(_f32,   f32,   7.0f32, 33.0f32, false, false, true, true);
+gen_tests!(_f64,   f64,   7.0f64, 33.0f64, false, false, true, true);
+gen_tests!(_char,  char,  'x',    '♥',     false, false, true, true);
+gen_tests!(_ptr, *mut String, 0 as *mut String, 0xBADC0DE as *mut String, false, false, false, false);
+gen_tests!(custom, super::Foo, super::Foo::Nothing, super::Foo::Set(0b101), false, false, true, false);
```

Comment:
These two lines are longer than 100 chars. Could you maybe do the following?

- Wrap all flags for the macro (logic, integer, default, serde) into `[ ]` just to make the invocation a bit clearer. So that we have `gen_tests!(_bool,  bool,  true,   false,   [true,  false, true, true]);` for example. 
- Secondly, remove the commas in the `[]`. `[true false true true]` is still fine for the macro.
- As a second step, replace `true` with `y` and `false` with `n`. The macro does not care what identifiers we use and `y`/`n` a short. The "header" in the comment that currently says `logic  int    dflt  serde` can just list them as `[logic int default serde]` now, even if it doesn't align anymore.

With those changes, this part of the code should be easier to read and the lines should be shorter.

If this explanation does not make sense to you, just let me know and I can push a commit that changes this.